src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/WorkspaceProjectFactoryService.cs

```diff
@@ -20,7 +20,9 @@ internal class WorkspaceProjectFactoryService : IWorkspaceProjectFactoryService,
     private readonly LanguageServerWorkspaceFactory _workspaceFactory;
     private readonly ProjectInitializationHandler _projectInitializationHandler;
 
+#pragma warning disable RS0030 // Do not use banned APIs (Required to match 'ExportBrokeredService')
```

Comment:
Should we just unban ImportingConstructor? You can't use it by itself without there also being an Export attribute that would similarly be blocked, except in cases like this where you have no alternative.

Reply:
➡️ Left this one as-is for now (could be addressed separately)

Reply:
Please still consider this; I'm not sure just moving the pragmas around is really an improvement when we can just make it work properly.

Reply:
➡️ Resolved by removing System.ComponentModel.Composition.ImportingConstructorAttribute from the banned symbol list.